### PR TITLE
Update ZeroMQ

### PR DESCRIPTION
**Pull request by str4d**
Includes changes cherry-picked from the following upstream PRs:

- bitcoin/bitcoin#9254
- bitcoin/bitcoin#13578
- bitcoin/bitcoin#15844
- bitcoin/bitcoin#16370
  - Only the ZeroMQ changes.
- bitcoin/bitcoin#16949

### CODE REVIEW

**Review by defuse — APPROVED** [review]
utACK

File: depends/packages.md [hunk 1]
```diff
@@ -146,3 +150,34 @@ $($(package)_config_opts) will be appended.
...
+
+From the [Gentoo Wiki entry](https://wiki.gentoo.org/wiki/Project:Quality_Assurance/Handling_Libtool_Archives):
+
+>  Libtool pulls in all direct and indirect dependencies into the .la files it
+>  creates. This leads to massive overlinking, which is toxic to the Gentoo
+>  ecosystem, as it leads to a massive number of unnecessary rebuilds.
+
+## Secondary dependencies:
+
+Secondary dependency packages relative to the bitcoin binaries/libraries (i.e.
```
Comment (daira) [inline depends/packages.md:168]:
```suggestion
Secondary dependency packages relative to the Zcash binaries/libraries (i.e.
```

File: depends/packages.md [hunk 2]
```diff
@@ -146,3 +150,34 @@ $($(package)_config_opts) will be appended.
...
+those not in `ALLOWED_LIBRARIES` in `contrib/devtools/symbol-check.py`) don't
+need to be shared and should be built statically whenever possible. This
+improves general build reliability as illustrated by the following example:
```
Comment (daira) [inline depends/packages.md:171]:
```suggestion
those not in `ALLOWED_LIBRARIES` in `contrib/devtools/symbol-check.py`)
should be built statically. The arguments for statically linking secondary
dependencies are similar to those for primary dependencies: it is preferable
for promptness of security updates; consensus compatibility; ease of
debugging and reproduction of user issues; avoiding unintended breakage
due to incompatible changes; and portability across OS distributions. It also
improves general build reliability as illustrated by the following example:
```

File: depends/packages.md [hunk 3]
```diff
@@ -5,6 +5,10 @@ The package "mylib" will be used here as an example
 
 General tips:
 - mylib_foo is written as $(package)_foo in order to make recipes more similar.
+- Secondary dependency packages relative to the bitcoin binaries/libraries (i.e.
+  those not in `ALLOWED_LIBRARIES` in `contrib/devtools/symbol-check.py`) don't
+  need to be shared and should be built statically whenever possible. See
+  [below](#secondary-dependencies) for more details.
```
Comment (daira) [inline depends/packages.md:11]:
```suggestion
- Secondary dependency packages relative to the Zcash binaries/libraries (i.e.
  those not in `ALLOWED_LIBRARIES` in `contrib/devtools/symbol-check.py`) should
  be built statically. See [below](#secondary-dependencies) for more details.
```

**Review by daira — APPROVED** [review]
utACK with suggestions.

**Comment by str4d** [comment]
@zkbot r+

**Comment by str4d** [comment]
Failure was due to CI internal failures that I hope are transient.

@zkbot retry

**Pull request merged**